src/frontends/jax/src/translate_session.cpp

```diff
@@ -0,0 +1,364 @@
+            }
+            // Add all tensors that were frozen
+            for (auto& desc : input_model->m_descriptors) {
+                (*tensor_map)[desc.first] = desc.second.m_value;
+            }
+        } else {
+            // Go over all jax_model inputs and register them in the tensor map:
+            auto inputs = jax_model->inputs();
+            for (size_t i = 0; i < inputs.size(); ++i) {
+                element::Type type = element::dynamic;
+                PartialShape pshape = jax_model->get_input_shape(i);
+                auto type_any = simplified_type_interpret(jax_model->get_input_type(i));
+                // TODO: Use special API to set custom type specification
+                if (type_any.is<element::Type>()) {
+                    type = type_any.as<element::Type>();
+                }
+                auto parameter = std::make_shared<v0::Parameter>(type, pshape);
+                parameter->set_friendly_name(jax_model->get_input_signature_name(i));
+                encode_tensor_name(parameter->output(0), inputs.at(i), {jax_model->get_input_signature_name(i)});
+                parameters->push_back(parameter);
+                (*tensor_map)[inputs.at(i)] = parameter;
+            }
+        }
+
+        auto node_visitor = [&](std::shared_ptr<JaxDecoder> node) {
+            // Explore all inputs of node. Node may refer to input value that hasn't
+            // been created in the current scope. But this value can be found in the
+            // outer scope, for this purpose we create new input for the model to link
+            // with external scope on a higher level.
+
+            auto raw_inputs = node->inputs();
+            for (size_t i = 0; i < raw_inputs.size(); ++i) {
+                auto input = raw_inputs.at(i);
+                if (tensor_map->find(input) == tensor_map->end()) {
+                    // Input refers value in the outer scope, need to create a new
+                    // Parameter in the current scope Linkage to external scope will be
+                    // performed on the level of the parent operation (if or loop)
+                    // TODO: Eliminate duplication with the main code for Parameters
+                    // creation
+                    PartialShape ps = node->get_input_shape(i);
+                    auto type = simplified_type_interpret(node->get_input_type(i));
+                    auto dtype = element::dynamic;
+                    if (type.is<element::Type>()) {
+                        dtype = type.as<element::Type>();
+                    }
+                    auto parameter = std::make_shared<v0::Parameter>(dtype, ps);
+                    (*tensor_map)[input] = parameter;
+                    // set name of parameter to the index of node in the model
+                    encode_tensor_name(parameter->output(0), input);
+                    parameters->push_back(parameter);
+                    inserted_params.push_back(input);
+                }
```

Comment:
Please remove this, this is also about nested model scope

Reply:
This part of code seems to be needed in JAX frontend except `inserted_params`, which has been removed.